tiledb/sm/rest/rest_client.h

```diff
@@ -32,6 +32,7 @@
 
 #ifndef TILEDB_REST_CLIENT_H
 #define TILEDB_REST_CLIENT_H
+#define REDIRECTION_HEADER_KEY "x-tiledb-full-region-domain"
```

Comment:
This is totally fine to hard-code, but let's use a C++ constant rather than a C-style definition.

Define this in `tiledb/sm/misc/constants.h/cc` instead.